core/designsystem/src/main/java/com/google/samples/apps/nowinandroid/core/designsystem/component/DynamicAsyncImage.kt

```diff
@@ -34,11 +42,29 @@ fun DynamicAsyncImage(
     placeholder: Painter? = null,
 ) {
     val iconTint = LocalTintTheme.current.iconTint
-    AsyncImage(
-        placeholder = placeholder,
+    SubcomposeAsyncImage(
+        error = {
+            if (placeholder != null) {
+                Image(
+                    painter = placeholder,
+                    contentDescription = "placeholder image",
+                )
+            }
+        },
```

Comment:
Not sure if this makes a difference, but the `if (placeholder != null)` could be lifted out, as there is no need to create the lambda if there is no `placeholder`.

Reply:
they always pass a placeholder, don't think this will male an difference

Reply:
That is not true, `InterestsIcon` and `TopicHeader` do not.

Reply:
yes your'e right, done